database.py

```diff
@@ -88,13 +92,51 @@ def get_trusted(self, userid: int) -> bool:
 			c.execute('''SELECT trusted FROM users WHERE userid = ?''', (userid,))
 			return bool(c.fetchone()[0])
 
+	def add_vk_messages(self, bad_user: int, msg_ids: list[int]):
+		'''
+		Adds `msg_ids` to the list of messages associated
+		with `bad_user`'s votekick.
+		'''
+		with self.mutex:
+			for msg_id in msg_ids:
+				self.db.execute(
+					'''INSERT INTO vk_messages VALUES (?, ?)''',
+					(bad_user, msg_id)
+				)
+			self.db.commit()
+
 	def cleanup_votekicks(self):
+		'''
+		Cleans up outdated votekick data from the database
+		'''
 		with self.mutex:
 			c = self.db.cursor()
 			c.execute('''DELETE FROM votekicks WHERE timeout < JULIANDAY('NOW')''')
 			c.fetchall()
 			self.db.commit()
 
+	def get_expired_messages(self) -> list[int]:
+		'''
+		Returns the list of messages to be deleted
+		and removes them from the database
+		'''
+		with self.mutex:
+			c = self.db.cursor()
+			c.execute('''
+				SELECT msg_id FROM vk_messages
+				WHERE bad_user NOT IN (
+					SELECT bad_user FROM votekicks
+				);''')
+			msgs = [row[0] for row in c.fetchall()]
+			c.execute('''
+				DELETE FROM vk_messages
+				WHERE bad_user NOT IN (
+					SELECT bad_user FROM votekicks
+				);''')
+			c.fetchall()

```

Comment:
just noticed, what is this second fetchall for? I don't think it's necessary for syncing the DB, is it? commit should already do that.